src/HotChocolate/Core/src/Execution/Pipeline/OperationComplexityMiddleware.cs

```diff
@@ -115,17 +116,14 @@ public async ValueTask InvokeAsync(IRequestContext context)
 
             foreach (var analyzer in analyzers)
             {
-                if (analyzer.OperationDefinitionNode == operationDefinition)
+                if (analyzer.OperationDefinitionNode.Equals(operationDefinition, SyntaxComparison.Syntax))
                 {
                     operationAnalyzer = analyzer.Analyzer;
-                }
 
-                _cache.TryAddAnalyzer(
-                    requestContext.CreateCacheId(
-                        CreateOperationId(
-                            requestContext.DocumentId!,
-                            analyzer.OperationDefinitionNode.Name?.Value)),
-                    analyzer.Analyzer);
+                    _cache.TryAddAnalyzer(
```

Comment:
This is incorrect, we create all analyzers here not just for the current operation. This should in the very least have a `break`. What is your reasoning here?